src/nix/modules/service/nixos-init.nix

```diff
@@ -0,0 +1,39 @@
+/*
+  Invokes the NixOS init system in the container.
+ */
+{ config, lib, pkgs, ... }:
+let
+  inherit (lib) types;
+in
+{
+  options = {
+    nixos.useInit = lib.mkOption {
```

Comment:
I'd call this `useSystemd`  and have different options if we decide to support a different init in the future.